test/test_ops.py

```diff
@@ -471,8 +471,7 @@ def _to_tensormeta(x):
         for ei in error_inputs:
             si = ei.sample_input
             meta_sample = si.transform(_to_tensormeta)
-            # TODO: match strings
-            with self.assertRaisesRegex(ei.error_type, ""):
```

Comment:
This is the main change, which required me to adjust everything else. Without this, strings of error inputs for refs are not checked.